### PR TITLE
Support publish buffer throttle for kop

### PR DESCRIPTION
**Pull request by aloyszhang**
#222 
Support publish buffer throttle for kop .

### CODE REVIEW

**Comment by aloyszhang** [comment]
@BewareMyPower Is ther any way to re-run the checks?

**Comment by BewareMyPower** [comment]
As far as I know there's not any script like `pulsarbot`. It requires a committer to manually rerun the workflow.

**Comment by BewareMyPower** [comment]
By the way, it's better to run the simple tests in your local environment before pushing the commits:

```bash
$ mvn checkstyle:check
$ mvn spotbugs:check
# The following tests may take a few time, you can rely the CI to detect errors
$ mvn test -DfailIfNoTests=false '-Dtest=!KafkaIntegrationTest,!DistributedClusterTest'
$ mvn test '-Dtest=KafkaIntegrationTest' -pl tests
```

**Comment by aloyszhang** [comment]
@BewareMyPower Thanks for you reply.  I'll check style and spotbug locally first.

**Comment by BewareMyPower** [comment]
Could you add a unit test to verify the throttle like [MessagePublishThrottlingTest.java](https://github.com/apache/pulsar/pull/3986/files#diff-27c437c82514b1c1bc5f1f909b0c7e71eb5d8571ac728103157d62621de154ac)? In addition, the Pulsar version is 2.6.2 now, please rebase to the latest commit.

**Comment by aloyszhang** [comment]
@BewareMyPower It's hard to test in kop for this feature since the the logic of publish buffer throttling are implement in Pulsar.

**Comment by aloyszhang** [comment]
Addition,   we can not use `BrokerService#checkMessagePublishBuffer()` in kop.

File: tests/src/test/java/io/streamnative/pulsar/handlers/kop/MessagePublishBufferThrottle.java [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+        try {
+            producer.send(new ProducerRecord<>(topic, new byte[1024])).get(1, TimeUnit.SECONDS);
+            Assert.fail("Send operation should failed because of timeout.");
+        } catch (TimeoutException e) {
+            // Expected
+        }
+
+
+        internalServerCnx.setMessagePublishBufferSize(0);
+        Thread.sleep(10000);
```
Comment (BewareMyPower) [inline tests/src/test/java/io/streamnative/pulsar/handlers/kop/MessagePublishBufferThrottle.java]: You should add comments here to indicate `sleep` is just for waiting until `checkMessagePublishBuffer` was called. Here's an example: https://github.com/apache/pulsar/blob/3142456107e8bc5d3d0d7fa526fda114c0d64c42/pulsar-broker/src/test/java/org/apache/pulsar/client/api/MessageDispatchThrottlingTest.java#L262

And use `conf.getMessagePublishBufferCheckIntervalInMillis()` here is more readable than hard coded 10000. By the way, IMO, just keep the default value (100) here is enough. You can just wait a little longer if you worried about the CI failure.

**Review by BewareMyPower — CHANGES_REQUESTED** [review]

File: tests/src/test/java/io/streamnative/pulsar/handlers/kop/MessagePublishBufferThrottle.java (hunk 1 above)
Comment (aloyszhang) [reply tests/src/test/java/io/streamnative/pulsar/handlers/kop/MessagePublishBufferThrottle.java]: you are right

**Review by BewareMyPower — APPROVED** [review]

**Pull request merged**